rest/src/main/java/io/syndesis/rest/v1/handler/integration/IntegrationHandler.java

```diff
@@ -108,6 +121,44 @@ public Integration get(String id) {
         return integration;
     }
 
+
+    @GET
+    @Path("/{id}/export.zip")
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public StreamingOutput export(@NotNull @PathParam("id") @ApiParam(required = true) String id) throws IOException {
+        ArrayList<ModelData> models = new ArrayList<>();
+
+        Integration integration = this.get(id);
+        models.add(new ModelData(Kind.Integration, integration));
+
+        for (Step step : integration.getSteps()) {
+            Optional<Connection> c = step.getConnection();
+            if( c.isPresent() ) {
+                Connection connection = c.get();
+                models.add(new ModelData(Kind.Connection, connection));
+                Connector connector = getDataManager().fetch(Connector.class, connection.getConnectorId().get());
+                if( connector != null ) {
+                    models.add(new ModelData(Kind.Connector, connector));
```

Comment:
Shouldn't you add a connector only if not already adedd ? (e.g. when you use the same connector but different actions for incoming and outcoming)

Reply:
Once again there is no harm in exporting the same thing multiple times.  It only gets imported once.